components/eamxx/src/share/io/scorpio_input.hpp

```diff
@@ -103,10 +103,15 @@ class AtmosphereInput
   // Cleans up the class
   void finalize();
 
+  // Getters
+  std::string get_filename() { return m_filename; } // Simple getter to query the filename for this stream.
+
+  // Expose the ability to set field manager for cases like time_interpolation where we swap fields
+  // between field managers to avoid deep_copy.
+  void set_field_manager (const std::shared_ptr<const fm_type>& field_mgr);
```

Comment:
If you want to allow to reset a FM, we should add some checks, to ensure that the fields in the new FM have the same layout (across all ranks). If accidentally they don't, we'd end up with reading errors, since the data structure in scorpio have already been set (e.g., the dofs offsets have already been set).